src/app/wallet/receive/modals/add-address-label/add-address-label.component.ts

```diff
@@ -43,7 +43,11 @@ export class AddAddressLabelComponent implements OnInit {
   onSubmitForm(): void {
     if (this.rpc.state.get('locked')) {
       // unlock wallet
-      this._modals.open('unlock', {forceOpen: true, timeout: 3, callback: this.addNewLabel.bind(this)});
+      if (this.type === 'private') {
+        this._modals.open('unlock', {forceOpen: true, timeout: 3, callback: this.addNewLabel.bind(this)});
+      } else {
```

Comment:
Adding public addresses also requires unlock

Reply:
I guess no ...

Reply:
I got this confused with the address book manager, seems like getting new private addresses does require an unlock.